### PR TITLE
Pytest Style analysis/test_base.py

### PR DESCRIPTION
**Pull request by utkbansal**
Fixes #

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: testsuite/MDAnalysisTests/analysis/test_base.py [hunk 1]
```diff
@@ -61,51 +60,49 @@ def _single_frame(self):
...
 
 
-class TestAnalysisBase(TestCase):
-    def setUp(self):
-        # has 98 frames
-        self.u = mda.Universe(PSF, DCD)
-
-    def tearDown(self):
-        del self.u
+class TestAnalysisBase(object):
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_base.py]: no need to have this as a class actually

File: testsuite/MDAnalysisTests/analysis/test_base.py [hunk 2]
```diff
@@ -61,51 +60,49 @@ def _single_frame(self):
...
+        return mda.Universe(PSF, DCD)
 
-    def test_default(self):
-        an = FrameAnalysis(self.u.trajectory).run()
-        assert_equal(an.n_frames, len(self.u.trajectory))
-        assert_equal(an.frames, list(range(len(self.u.trajectory))))
+    def test_default(self, u):
+        an = FrameAnalysis(u.trajectory).run()
+        assert an.n_frames == len(u.trajectory)
+        assert_equal(an.frames, list(range(len(u.trajectory))))
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_base.py]: assert_array_equal

Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_base.py]: please also change for similar cases in this file

File: testsuite/MDAnalysisTests/analysis/test_base.py [hunk 3]
```diff
@@ -139,12 +137,9 @@ def simple_function(mobile):
 def test_AnalysisFromFunction():
     u = mda.Universe(PSF, DCD)
     step = 2
-    ana1 = base.AnalysisFromFunction(simple_function, mobile=u.atoms,
-                                     step=step).run()
-    ana2 = base.AnalysisFromFunction(simple_function, u.atoms,
-                                     step=step).run()
-    ana3 = base.AnalysisFromFunction(simple_function, u.trajectory, u.atoms,
-                                     step=step).run()
+    ana1 = base.AnalysisFromFunction(simple_function, mobile=u.atoms, step=step).run()
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_base.py]: please don't change this formatting. I'm sure this was done to conform to the 80 char limit of pep8

File: testsuite/MDAnalysisTests/analysis/test_base.py (hunk 2 above)
Comment (jbarnoud) [inline testsuite/MDAnalysisTests/analysis/test_base.py]: See discussion on #1444.

Comment (utkbansal) [inline testsuite/MDAnalysisTests/analysis/test_base.py]: @kain88-de my understanding is that `assert_equal` is preferred over `assert_array_equal` as discussed in 1444?

Comment (jbarnoud) [inline testsuite/MDAnalysisTests/analysis/test_base.py]: @utkbansal It is my understanding too.

**Comment by kain88-de** [comment]
coverage decreased

**Comment by utkbansal** [comment]
@kain88-de Can't figure the drop in coverage, restarting the build.

**Comment by utkbansal** [comment]
@kain88-de @jbarnoud @richardjgowers There is something fishy about the coverage. I have the same 4 lines losing coverage in #1541 too.

**Comment by richardjgowers** [comment]
@utkbansal which 4 lines?

**Comment by utkbansal** [comment]
@richardjgowers 2 lines in `core/topologyattrs.py` and 2 lines in `lib/transformations.py`. Have a look at the coverage report at https://coveralls.io/builds/12561272

I have seen the 2 lines in `lib/transformations.py` loose and gain coverage unpredictably in previous PRs too, but restarting the build usually fixed it.

**Comment by utkbansal** [comment]
@richardjgowers @kain88-de What should I do about the drop?

**Pull request merged**